subprojects/core/src/main/java/org/gradle/api/artifacts/transform/internal/ArtifactTransforms.java

```diff
@@ -82,26 +84,47 @@ public DependencyTransformRegistration(String from, String to, Class<? extends A
         public Transformer<File, File> getTransformer() {
             ArtifactTransform artifactTransform = DirectInstantiator.INSTANCE.newInstance(type);
             config.execute(artifactTransform);
-            return new DependencyTransformTransformer(artifactTransform, outputProperty);
+            return new DependencyTransformTransformer(artifactTransform, outputProperty, to);
         }
     }
 
     private static class DependencyTransformTransformer implements Transformer<File, File> {
+        private static final Logger LOGGER = Logging.getLogger(DependencyTransformTransformer.class);
+
         private final ArtifactTransform artifactTransform;
         private final JavaMethod<? super ArtifactTransform, File> outputProperty;
+        private final String outputFormat;
 
-        private DependencyTransformTransformer(ArtifactTransform artifactTransform, JavaMethod<? super ArtifactTransform, File> outputProperty) {
+        private DependencyTransformTransformer(ArtifactTransform artifactTransform, JavaMethod<? super ArtifactTransform, File> outputProperty, String outputFormat) {
             this.artifactTransform = artifactTransform;
             this.outputProperty = outputProperty;
+            this.outputFormat = outputFormat;
         }
 
         @Override
         public File transform(File file) {
             if (artifactTransform.getOutputDirectory() != null) {
                 artifactTransform.getOutputDirectory().mkdirs();
             }
-            artifactTransform.transform(file);
-            return outputProperty.invoke(artifactTransform);
+            File output = null;
+            try {
+                artifactTransform.transform(file);
+                output = outputProperty.invoke(artifactTransform);
+                if (output == null) {
+                    reportError(file, "no output file created", null);
+                } else if (!output.exists()) {
+                    reportError(file, "expected output file '" + output.getPath() + "' was not created", null);
+                }
+            } catch (Exception e) {
+                reportError(file, e.getMessage(), e);
+            }
+            return output;
+        }
+
+        private void reportError(File input, String message, Exception e) {
+            LOGGER.error(String.format("Error while transforming '%s' to format '%s' using '%s'%s",
```

Comment:
We now throw an exception in `internal.ArtifactTransforms`. Is that fine or do we want to move that further up to have a _cleaner_ stack trace (with less internal/private methods)? Is that something we are concerned about?

I used a `FileNotFoundException` in addition if the output is missing. I did that to get the nice hierarchical _cause_ formatting in the problem report. I.e.:

```
> Error while transforming 'guava-19.0.jar' to format 'md5' using 'ToNullTransform'
   > Expected output file 'this/file/does/not/exist' was not created
```

Is that how it should be done?